### DISCUSSION TITLE
No sound out of the GY PCM5102 DAC module

### DISCUSSION THREAD

**Discussion by atulrnt** [probonopd/MiniDexed#653] [Q&A] [answered]
Hi,

I've been desperately trying to get some sound coming out of the GY PCM5102 DAC module without success.
It seems to me I've soldered and connected everything correctly (jumpers and module to the raspberry).

I'm using a Raspberry PI 3B+ and I have sound coming out of the onboard jack when `SoundDevice` is set to `pwm`.
I ordered 3 DACs just in case, I tried all of them and the result is the same so it's not a board issue I think.

I've connected everything as described here : https://github.com/dwhinham/mt32-pi/wiki/GY-PCM5102-DAC-module
Which describes the same connections as the diagram from MiniDexed.
![image](https://github.com/probonopd/MiniDexed/assets/894203/f98a7135-8c60-4fb5-9d56-b791167a2acc)

As you can see, I've configured minidex.ini to use `i2s` for the `SoundDevice`.
![image](https://github.com/probonopd/MiniDexed/assets/894203/353248af-dd30-4616-914c-f981d25d2772)

The jumpers are correctly soldered.
![IMG_20240506_145110](https://github.com/probonopd/MiniDexed/assets/894203/28d37ba3-49af-4428-8f1f-19c3251a3196)

The power led is ON.
![IMG_20240506_145034](https://github.com/probonopd/MiniDexed/assets/894203/5b000237-5c0d-4862-83c2-f87f5aad6354)

Here is the boot output.
![IMG_20240506_144949](https://github.com/probonopd/MiniDexed/assets/894203/9224bdb3-f75c-4f2c-829b-3d490e416a84)

Am I missing anything?

**Comment by diyelectromusic** [comment]
I can't see anything obviously wrong I'm afraid.  For completeness, the RPi GPIO connections for the PCM5102 are:
18 - BCK
19 - LCK
21 - DIN

And they are usually powered from 5V not 3V3.

... but I dare say you've checked those a few times already :/

I'm struggling to think about what to try next tbh.  Can you check the modules with something other than MiniDexed?

I see the compile time on your build was Jan 2024.  Grasping at straws, and I've no reason to believe there was an issue with that, but is it worth trying the latest build?

Other than that, it is down to the "silly" things - things like making sure the jack lead/amplification is fine - that kind of thing (I once thought one was failing and realised I'd left the jack in the Pi's socket not the PCM5102!).  These are stereo sockets, so making sure you have a stereo to stereo connection - plugging these into a mono amp will probably short the signal - that kind of thing...?  But if PWM is working, that's all probably fine...

As I say, I'm really scrapping around for something to think about I'm afraid.

Sorry, at the moment, I can't be of more help.

Kevin

**Comment by probonopd** [comment]
Can you get sound out when using Raspberry Pi OS?

**Reply by atulrnt** [reply]
I'm not at home at the moment but will the DAC be recognized directly or do I need to do anything for it to work in the PI OS?

**Reply by probonopd** [reply]
You will need to do something (e.g., configure the overlay for it).

**Comment by Lonolo** [comment]
Have you tried different 1/8" stereo (TRS only) audio cables too?

**Comment by atulrnt** [comment, marked as answer]
I completely erased the micro SD card and reinstalled the latest version of minidexed and it's now working!

What a difference between the pwm and the DAC output!

Thank you for your help!

**Reply by Banana71** [reply]
If only the price-performance ratio was as good everywhere as in this project .... 
Yes, this DAC sounds really good.

We've had this happen several times now that there were random problems that worked by reformatting and reinstalling the system on the micro SD card. 🤷🏼‍♂️
Nice Case 👍🏼

**Comment by atulrnt** [comment]
And here it is in its final form, thank you for this excellent project!!

![IMG_20240507_120400](https://github.com/probonopd/MiniDexed/assets/894203/528b32e7-64fa-4567-af50-c33ffe7d6299)
